testdata/p4_16_samples/psa-i2e-cloning-basic-bmv2.p4

```diff
@@ -0,0 +1,148 @@
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+*/
+
+#include <core.p4>
+#include "psa.p4"
+
+
+typedef bit<48>  EthernetAddress;
+
+header ethernet_t {
+    EthernetAddress dstAddr;
+    EthernetAddress srcAddr;
+    bit<16>         etherType;
+}
+
+struct empty_metadata_t {
+}
+
+struct metadata_t {
+}
+
+struct headers_t {
+    ethernet_t       ethernet;
+}
+
+parser IngressParserImpl(packet_in pkt,
+                         out headers_t hdr,
+                         inout metadata_t user_meta,
+                         in psa_ingress_parser_input_metadata_t istd,
+                         in empty_metadata_t resubmit_meta,
+                         in empty_metadata_t recirculate_meta)
+{
+    state start {
+        pkt.extract(hdr.ethernet);
+        transition accept;
+    }
+}
+
+control cIngress(inout headers_t hdr,
+                 inout metadata_t user_meta,
+                 in    psa_ingress_input_metadata_t  istd,
+                 inout psa_ingress_output_metadata_t ostd)
+{
+    action clone() {
+        ostd.clone = true;
+        ostd.clone_session_id = (CloneSessionId_t) (CloneSessionIdUint_t) 8;
+    }
+
+    apply {
+        clone();
+
+        if (hdr.ethernet.dstAddr == (EthernetAddress) 9) {
+            // PROPERTY - dropped packets are still cloned
+            ingress_drop(ostd);
+        } else {
+            // mark the original packet so stf can tell the difference btwn this and clones
+            hdr.ethernet.srcAddr = (EthernetAddress) 51966;
+            // GENERAL CASE - all other packets are sent to port number from packet
+            send_to_port(ostd, (PortId_t) (PortIdUint_t) hdr.ethernet.dstAddr);
+        }
+    }
+}
+
+parser EgressParserImpl(packet_in buffer,
+                        out headers_t hdr,
+                        inout metadata_t user_meta,
+                        in psa_egress_parser_input_metadata_t istd,
+                        in empty_metadata_t normal_meta,
+                        in empty_metadata_t clone_i2e_meta,
+                        in empty_metadata_t clone_e2e_meta)
+{
+    state start {
+        buffer.extract(hdr.ethernet);
+        transition accept;
+    }
+}
+
+control cEgress(inout headers_t hdr,
+                inout metadata_t user_meta,
+                in    psa_egress_input_metadata_t  istd,
+                inout psa_egress_output_metadata_t ostd)
+{
+    apply {
+        // mark the clone packets so stf can tell the difference
+        if (istd.packet_path == PSA_PacketPath_t.CLONE_I2E) {
+            hdr.ethernet.srcAddr = 48w5;
+        }
```

Comment:
@peteli3 Sorry for one more request on changes to this test program.  Right now this test program modifies hdr.ethernet.srcAddr on ingress, which should not affect clone packets, but it also assigns a value to the same field in egress, only for cloned packets, so from the output packet it isn't really possible to tell that the ingress assignment DIDN'T take effect, like it should not.

Could you change either the ingress or the egress assignment to a different field, e.g. hdr.ethernet.etherType?  That way whichever field you try to assign in ingress, that does not come out modified for cloned packets, we can know that it is not being modified at all by the P4 program, as it should not.

Reply:
That's a good point - thanks for the suggestion. I'll make this change.

Reply:
the next commit adds this suggested change